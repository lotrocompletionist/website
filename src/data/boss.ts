export interface IBoss {
  id: number;
  name: string;
}
