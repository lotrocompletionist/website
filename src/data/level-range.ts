export interface ILevelRange {
  minimum: number;
  maximum?: number;
}
