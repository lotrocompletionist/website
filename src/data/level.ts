export interface Level {
  minimum: number;
  maximum?: number;
}
