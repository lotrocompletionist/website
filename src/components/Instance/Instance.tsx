import * as React from "react";
import {
  CardContent,
  Media,
  MediaLeft,
  MediaContent,
  Content,
  Image,
  Title,
  Subtitle
} from "bloomer";
import { Instance } from "../../data/models";
import { Level } from "../Level/Level";

export interface InstanceProps {
  instance: Instance;
}

export function Instance(props: InstanceProps) {
  return (
    <>
      <CardContent>
        <Media>
          <MediaLeft>
            <Image isSize="48x48" src="https://via.placeholder.com/96x96" />
          </MediaLeft>
          <MediaContent>
            <Title isSize={4}>{props.instance.name}</Title>
            <Subtitle isSize={6}>
              {props.instance.abbreviations.join(", ")}
            </Subtitle>
          </MediaContent>
        </Media>
        <Content>
          {props.instance.cluster}
          <br />
          <small>
            <Level level={props.instance.level} />
          </small>
        </Content>
      </CardContent>
    </>
  );
}
