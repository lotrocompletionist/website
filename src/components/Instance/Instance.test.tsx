import * as React from "react";
import { create } from "react-test-renderer";
import { Instance } from "./Instance";
import { IInstance } from "../../data/instance";

test("<Instance> renders correctly", () => {
  const instance: IInstance = {
    id: 1,
    name: "Great Barrow: The Maze",
    abbreviations: ["Maze"],
    region: "Bree-land",
    level: { minimum: 20, maximum: 115 },
    tiers: 1,
    groups: ["Fellowship"],
    availability: ["Free"],
    challenge: "Daily",
    cluster: "Shadows of Angmar",
    note: "Located in Southern Barrow-downs, see also the Great Barrow",
    bosses: [
      { id: 1, name: "Barrow-wight Caller" },
      { id: 2, name: "Gaerdring" },
      { id: 3, name: "Gaerthel" }
    ]
  };
  const tree = create(<Instance instance={instance} />).toJSON();
  expect(tree).toMatchSnapshot();
});
