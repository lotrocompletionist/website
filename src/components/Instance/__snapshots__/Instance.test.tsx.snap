// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Instance> renders correctly 1`] = `
<div
  className="card-content"
>
  <article
    className="media"
  >
    <div
      className="media-left"
    >
      <figure
        className="image is-48x48"
      >
        <img
          src="https://via.placeholder.com/96x96"
        />
      </figure>
    </div>
    <div
      className="media-content"
    >
      <h1
        className="title is-4"
      >
        Great Barrow: The Maze
      </h1>
      <h2
        className="subtitle is-6"
      >
        Maze
      </h2>
    </div>
  </article>
  <div
    className="content"
  >
    Shadows of Angmar
    <br />
    <small>
      Level: 
      20
       - 115
      <br />
      Bosses: 
      Barrow-wight Caller, Gaerdring, Gaerthel
    </small>
  </div>
</div>
`;
